frontends/precinct-scanner/src/screens/scan_warning_screen.tsx

```diff
@@ -111,33 +119,56 @@ function UndervoteWarningScreen({
         find(contests, (contest) => contest.id === undervote.contestId).title
     );
 
+  const partiallyVotedContestNamesDisplayMaxCount = 5;
+  const partiallyVotedContestNamesDisplayMinCount = 3;
+  const partiallyVotedContestNamesDisplayCount =
+    partiallyVotedContestNames.length >
+    partiallyVotedContestNamesDisplayMaxCount
+      ? partiallyVotedContestNamesDisplayMinCount
+      : partiallyVotedContestNames.length;
+  const partiallyVotedContestNamesDisplayRemainder =
+    partiallyVotedContestNames.length - partiallyVotedContestNamesDisplayCount;
+  const partiallyVotedContestNamesDisplay = partiallyVotedContestNames.slice(
+    0,
+    partiallyVotedContestNamesDisplayCount
+  );
+  if (partiallyVotedContestNamesDisplayRemainder) {
+    partiallyVotedContestNamesDisplay.push(
+      `${partiallyVotedContestNamesDisplayRemainder} more ${pluralize(
+        'contest',
+        partiallyVotedContestNamesDisplayRemainder
+      )}`
+    );
+  }
```

Comment:
does this logic need to apply to `blankContestNames` as well?

Reply:
Yes. Will create shared function.